### PR TITLE
IGNITE-22883 Send all cluster IDs to thin client on connection

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-22883

Whole history of cluster IDs is passed the client handler so it can pass it to clients when they connect to allow them do cluster ID validation, but support cluster ID change (due to CMG repair).

### CODE REVIEW

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClusterInfo.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public class ClusterInfo {
+    private final ClusterTag tag;
+    private final List<UUID> idHistory;
+
+    /**
+     * Constructor.
+     */
+    public ClusterInfo(ClusterTag tag, List<UUID> idHistory) {
+        this.tag = tag;
+        this.idHistory = List.copyOf(idHistory);
```
Comment (ptupitsyn) [inline modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClusterInfo.java:36]: I don't think we need a defensive copy in the internal API, the caller already creates a new `ArrayList`.

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java [hunk 2]
```diff
@@ -276,7 +275,7 @@ public void enable() {
      */
     private CompletableFuture<Channel> startEndpoint(ClientConnectorView configuration) {
         ServerBootstrap bootstrap = bootstrapFactory.createServerBootstrap();
-        CompletableFuture<ClusterTag> clusterTag = clusterTagSupplier.get();
+        CompletableFuture<ClusterInfo> clusterInfo = clusterInfoSupplier.get();
```
Comment (ptupitsyn) [inline modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java:278]: `startEndpoint` is called once during node startup. Looks like we won't receive any cluster id history updates that may occur during the node life time.

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java (hunk 2 above)
Comment (rpuch) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java]: Currently, this doesn't seem to be a problem as a cluster ID is always changed via a restart. But we now have a storage from which we can read ClusterState, allowing us to not use the future (and `join()`) anymore, so I switched to this approach.

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClusterInfo.java (hunk 1 above)
Comment (rpuch) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClusterInfo.java]: Fixed

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**